### PR TITLE
[Docs] Point collections info to docs.ansible.com and remove mazer

### LINKED ISSUE #2021: Update Collection related documentation
- [ ] Docs related to developing and interacting with Collections will live at docs.ansible.com, and the Galaxy docs will link users there. 
- [ ] Mazer has been deprecated and will be removed from Galaxy docs
- [ ] Any references or examples using `mazer` will be updated to reference `ansible-galaxy`
- [ ] Galaxy docs will point users to `ansible-galaxy` docs at docs.ansible.com

### PR DESCRIPTION
**Pull request by samccann**
Use collections documentation on docs.ansible.com for the details of how to create, upload, install, and use collections on galaxy. This provides one single source to maintain but keeps the information accessible from both docsites.

Removes mazer documentation and leaves stub files to point back to Ansible docs instead.

Fixes #2021

### CODE REVIEW

**Comment by samccann** [comment]
Still need to switch links for uploading a collection using galaxy UI and decide what to do about collections information under the mazer doc section...

File: docs/docsite/rst/contributing/creating_collections.rst [hunk 1]
```diff
@@ -4,178 +4,13 @@
...
-Collections
-===========
-
 Collections are a distribution format for Ansible content. They can be used to
-package and distribute playbooks, roles, modules, and plugins.
-
-.. note::
-    This is a Tech-Preview feature and is only supported by Ansible 2.8 (or greater).
-    Future Galaxy or Ansible releases may introduce breaking changes.
+package and distribute playbooks, roles, modules, and plugins. See  To learn how to develop collections, see  :ref:`Developing collections <ansible:developing_collections>`.
```
Comment (acozine) [inline docs/docsite/rst/contributing/creating_collections.rst]:
```suggestion
package and distribute playbooks, roles, modules, and plugins. To learn how to develop collections, see  :ref:`Developing collections <ansible:developing_collections>`.
```

File: docs/docsite/rst/finding/content_types.rst [hunk 2]
```diff
@@ -55,12 +55,8 @@ Ansible Collections
 ===================
 
 Ansible 2.8 introduced support for a new packaging format for creating and sharing Ansible content. This new format is called a Collection, and it enables the combining of
```
Comment (acozine) [inline docs/docsite/rst/finding/content_types.rst]:
```suggestion
Ansible 2.8 introduced support for a new packaging format for creating and sharing Ansible content. This new format is called a Collection, and it lets you combine
```

File: docs/docsite/rst/using/installing.rst [hunk 3]
```diff
@@ -238,59 +238,4 @@ the Ansible doc site <https://docs.ansible.com/ansible/latest/user_guide/playboo
...
-    │                       ├── README.md
-    │                       ├── meta
-    │                       │   └── main.yaml
-    │                       └── tasks
-    │                           └── main.yml
-
-Use the ``--collections-path`` option, or set the ``collections_path`` value in the
-``~/.ansible/mazer.yml`` config file to override the default install path. For more on installing, configuring,
-and using Mazer, view :doc:`../mazer/index`.
+See :ref:`Installing collections <ansible:collections>` for detailed information about installing collections in Galaxy.
```
Comment (acozine) [inline docs/docsite/rst/using/installing.rst]:
```suggestion
See :ref:`Installing collections <ansible:collections>` for detailed information about installing and using collections from Galaxy.
```

**Pull request merged**